cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -502,13 +505,11 @@ impl<I: VCodeInst> MachBuffer<I> {
     }
 
     /// Resolve a label to an offset, if known. May return `UNKNOWN_LABEL_OFFSET`.
-    fn resolve_label_offset(&self, label: MachLabel) -> CodeOffset {
-        let alias = self.label_aliases[label.0 as usize];
-        if alias != UNKNOWN_LABEL {
-            self.label_offsets[alias.0 as usize]
-        } else {
-            self.label_offsets[label.0 as usize]
+    fn resolve_label_offset(&self, mut label: MachLabel) -> CodeOffset {
+        while self.label_aliases[label.0 as usize] != UNKNOWN_LABEL {
```

Comment:
Is there anything you can do here to stop this looping infinitely in the case of incorrectly constructed `self.label_aliases`?  On the basis that an assertion failure, in the field, is more informative than a hang.  Like, for example, counting iterations and release-asserting at 1 million?  (I think RA only allows 1 million BBs anyway).

Reply:
Done!